### PR TITLE
feat: Universal snippet for django CMS 3 and 4

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

This PR updates django CMS snippet to become universal, i.e. support django CMS 3.11, 4.0, and 4.1. 

All features from the `support/django-cms-4.0.x` branch are included. 

* The data structure for django CMS 3 changes, and a `SnippetGrouper` model is introduced. This has no impact on end-users
* Snippets can be site-specific.
* Together with django CMS 4 and djangocms-versioning, snippets can be versioned.

This will be the base for version 5 of djangocms-snippet:
* Version <=3.x: django CMS 3.x
* Version <=4.x: django CMS 4.0/4.1 only
* Version >=5: Universal

## Related resources

* #...
* #...

## Checklist

* [x] I have opened this pull request against ``master``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by fsbraun** [comment]
@joshyu Can you have a look at this? It should also work with django CMS 4.0. I'd appreciate if you found time to test.

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 3.1.0
+current_version = 5.0.0a
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply setup.cfg:2]: That's the version for django CMS 4.0 compatible snippets in `support/django-cms-4.0.x`: https://github.com/django-cms/djangocms-snippet/blob/support/django-cms-4.0.x/djangocms_snippet/__init__.py

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -8,30 +8,18 @@ ci:
...
-     hooks:
-       -  id: pyupgrade
-          args: ["--py38-plus"]
-
-  - repo: https://github.com/adamchainz/django-upgrade
-    rev: '1.16.0'
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
     hooks:
-      - id: django-upgrade
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply .pre-commit-config.yaml]: Should not have been removed. It is back in now. It's just that the branches being merged here have quite a long separate history now...

File: setup.cfg [hunk 3]
```diff
@@ -21,3 +21,33 @@ replace =
 
 	{new_version} {utcnow:%%Y-%%m-%%d}
 	================
+
+[flake8]
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply setup.cfg]: Done 👍

**Comment by joshyu** [comment]
@fsbraun，
I add a comment in the issue above, I try to install snippet addon to djangocms-quickstart project, but failed to install with some error. Please help to check the message I left in the above issue?

**Comment by joshyu** [comment]
@fsbraun ,
Can we update the `MANIFEST.in` as below:
image

**Comment by fsbraun** [comment]
@joshyu Interesting since `MANIFEST.in` was not modified. It needs to be updated to also include template files. I'll do the update.

**Comment by joshyu** [comment]
@fsbraun , 

I think you also need to include `*.po` in the MANIFEST.in , otherwise the `locale` directory will be excluded from the setup package.

image

**Comment by fsbraun** [comment]
@joshyu Fixed.

**Pull request merged**